src/signature.sh

```diff
@@ -0,0 +1,348 @@
+        trailer_operations+=('ADD_FIXES')
+        shift 2
+        ;;
+      --verbose)
+        options_values['VERBOSE']=1
+        shift
+        ;;
+      --)
+        # End of options, beginning of arguments.
+        # Overwrite default value if at least one argument is given.
+        [[ -n "$2" ]] && options_values['PATCH_OR_SHA']=''
+        shift
+        ;;
+      *)
+        # Get all passed arguments each loop
+        if [[ "$1" == *"*"* ]]; then
+          # Expand the glob and loop through each resulting file
+          for arg in $1; do
+            options_values['PATCH_OR_SHA']+=" $arg"
+          done
+        else
+          options_values['PATCH_OR_SHA']+=" $1"
+        fi
+        shift
+        ;;
+    esac
+  done
+}
+
+function signature_help()
+{
+  if [[ "$1" == --help ]]; then
+    include "${KW_LIB_DIR}/help.sh"
+    kworkflow_man 'handle-trailer'
+    return
+  fi
+  printf '%s\n' 'kw handle-trailer:' \
+    '  handle-trailer (--add-signed-off-by | -s) (<name>) [<patchset> | <sha>] - Add Signed-off-by' \
+    '  handle-trailer (--add-reviewed-by | -r) (<name>) [<patchset> | <sha>] - Add Reviewed-by' \
+    '  handle-trailer (--add-acked-by | -a) (<name>) [<patchset> | <sha>] - Add Acked-by' \
+    '  handle-trailer (--add-fixes | -f) [<fixed-sha>] [<patchset> | <sha>] - Add Fixes' \
+    '  handle-trailer (--verbose) - Show a detailed output'
+}
+
+# This function registers a signature operation into
+# trailer_operations that will be properly excuted later.
+# It attempts to use the user's name and email configured
+# if a signature is not passed as argument and gives
+# an error if they are not properly set with git config.
+#
+# @keyword Holds a string that is a keyword for a specific operation.
+# @signature Holds a string like 'NAME <EMAIL>' defining the signature.
+function register_signature_operation()
+{
+  local keyword="$1"
+  local signature="$2"
+
+  local operation_key="ADD_${keyword}"
+  local signature_key="${keyword}_LINE"
+  local formated_output
+
+  options_values["$operation_key"]=1
+  options_values["$signature_key"]="${TRAILER_TAGS[$keyword]} "
+
+  # Use default from git config if no argument was given
+  if [[ ! "$signature" ]]; then
+    formated_output="$(format_name_email_from_user)"
+    if [[ "$?" -gt 0 ]]; then
+      return 1
+    fi
+    options_values["$signature_key"]+="$formated_output"
+  else
+    options_values["$signature_key"]+="$(str_strip "$signature")"
+  fi
+
+  trailer_operations+=("$operation_key")
+}
+
+# This function receives a commit SHA and verifies if it's
+# a valid commit reference. If it is, then it outputs the
+# appropriate formatted message. Else it returns an error code.
+#
+# @sha Holds either a commit hash or pointer
+function format_fixes_message()
+{
+  local sha="$1"
+  local formatted_message
+
+  # Check if given value is a valid commit reference
+  if [[ $(git cat-file -t "$sha" 2> /dev/null) != 'commit' ]]; then
```

Comment:
btw, it would be awesome if you could work on the below issue since you wrote a lot of git code:

https://github.com/kworkflow/kworkflow/issues/1123